addons/mass_mailing/models/mailing.py

```diff
@@ -837,6 +837,27 @@ def action_fetch_favorites(self, extra_domain=None):
 
         return values_list
 
+    def action_model_mass_mail(self, model_id, user, records):
+        """ Open a new mailing with the passed records as default target domain """
+        context = {
+            'default_reply_to_mode': 'new',
+            'default_mailing_model_id': model_id,
+        }
+        # check that user has mailing access ; if not, open the simplified multi-send composer
+        if not user.has_group("mass_mailing.group_mass_mailing_user"):
+            return self.env['ir.actions.act_window']._for_xml_id('mail.action_partner_mass_mail')
+        if records._name == 'mailing.list':
+            context['default_contact_list_ids'] = records.ids
+        else:
+            context['default_mailing_domain'] = "[('id', 'in', [" + ','.join(map(str, records.ids)) + '])]'
```

Comment:
Avoir manual stringification. You can just say json.dumps(my_list) (if I remember well json, not the kind of people using it regularly). Or repr(my_list), works too (json should be a bit more tailored for lists).